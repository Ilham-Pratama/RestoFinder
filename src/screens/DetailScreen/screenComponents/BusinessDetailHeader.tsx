import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import sharedStyles from 'shared/styles';
import Rating from 'components/Rating';
import ImageCarousel from 'components/ImageCarousel';
import { Business } from 'types/Business';

const BusinessDetailHeader = ({
  name,
  photos,
  rating,
  review_count
}: Business) => {
  return (
    <View>
      <ImageCarousel images={photos} />
      <LinearGradient
        // accessibilityRole="none"
        colors={[
          '#00000000',
          '#00000000',
          '#00000000',
          '#0000005d',
          '#000000a1',
          '#000000ce'
        ]}
        style={styles.darkBg}
      />
      <View
        style={[
          styles.absoluteContainer,
          sharedStyles.paddingVertical2,
          sharedStyles.paddingHorizontal3
        ]}>
        <Text style={styles.name} accessibilityRole="header">
          {name}
        </Text>
        <Rating rating={rating} review_count={review_count} />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  darkBg: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0
  },
  absoluteContainer: {
    position: 'absolute',
    justifyContent: 'flex-end',
    bottom: 0,
    left: 0,
    right: 0
  },
  name: {
    fontSize: 30,
    fontWeight: '800',
    color: '#f4f4f4'
  },
  reviewContainer: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  reviewCount: {
    color: '#f4f4f4'
  }
});

export default BusinessDetailHeader;
