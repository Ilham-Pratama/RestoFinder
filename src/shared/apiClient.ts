import axios from 'axios';
import { YELP_API_TOKEN } from 'react-native-dotenv';

export const BASE_API_URL = 'https://api.yelp.com/v3';

export const apiClient = axios.create({
  baseURL: BASE_API_URL,
  timeout: 1000,
  headers: {
    Authorization: `Bearer ${YELP_API_TOKEN}`
  }
});
