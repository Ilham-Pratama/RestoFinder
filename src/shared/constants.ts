import { ICategoryItem } from 'components/CategoryItem/types';

export const MENU_CATEGORIES: ICategoryItem[] = [
  {
    label: 'Burger',
    value: 'burgers',
    image: require('assets/images/burger.png')
  },
  {
    label: 'Dessert',
    value: 'desserts',
    image: require('assets/images/cake.png')
  },
  {
    label: 'Pasta',
    value: 'pastashops',
    image: require('assets/images/pasta.png')
  },
  {
    label: 'Pizza',
    value: 'pizza',
    image: require('assets/images/pizza.png')
  },
  {
    label: 'Smoothies',
    value: 'juicebars',
    image: require('assets/images/smoothies.png')
  },
  {
    label: 'steak',
    value: 'steak',
    image: require('assets/images/steak.png')
  }
];
