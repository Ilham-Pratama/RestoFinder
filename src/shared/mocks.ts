export const BUSINESS_DUMMY = {
  id: 'WavvLdfdP6g8aZTtbBQHTw',
  alias: 'gary-danko-san-francisco',
  name: 'Gary Danko',
  image_url:
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
  is_claimed: true,
  is_closed: false,
  url: 'https://www.yelp.com/biz/gary-danko-san-francisco?adjust_creative=wpr6gw4FnptTrk1CeT8POg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_lookup&utm_source=wpr6gw4FnptTrk1CeT8POg',
  phone: '+14157492060',
  display_phone: '(415) 749-2060',
  review_count: 5296,
  categories: [
    {
      alias: 'newamerican',
      title: 'American (New)'
    },
    {
      alias: 'french',
      title: 'French'
    },
    {
      alias: 'wine_bars',
      title: 'Wine Bars'
    }
  ],
  rating: 4.5,
  location: {
    address1: '800 N Point St',
    address2: '',
    address3: '',
    city: 'San Francisco',
    zip_code: '94109',
    country: 'US',
    state: 'CA',
    display_address: ['800 N Point St', 'San Francisco, CA 94109'],
    cross_streets: ''
  },
  coordinates: {
    latitude: 37.80587,
    longitude: -122.42058
  },
  photos: [
    'https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/FmXn6cYO1Mm03UNO5cbOqw/o.jpg',
    'https://s3-media4.fl.yelpcdn.com/bphoto/HZVDyYaghwPl2kVbvHuHjA/o.jpg'
  ],
  price: '$$$$',
  hours: [
    {
      open: [
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 0
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 1
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 2
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 3
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 4
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 5
        },
        {
          is_overnight: false,
          start: '1730',
          end: '2200',
          day: 6
        }
      ],
      hours_type: 'REGULAR',
      is_open_now: false
    }
  ],
  transactions: [],
  special_hours: [
    {
      date: '2019-02-07',
      is_closed: null,
      start: '1600',
      end: '2000',
      is_overnight: false
    }
  ]
};

export const REVIEW_DUMMY = {
  id: 'xAG4O7l-t1ubbwVAlPnDKg',
  rating: 5,
  user: {
    id: 'W8UK02IDdRS2GL_66fuq6w',
    profile_url:
      'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
    image_url:
      'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
    name: 'Ella A.'
  },
  text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
  time_created: '2016-08-29 00:41:13',
  url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
};

export const REVIEWS_DUMMY = {
  reviews: [
    {
      id: 'xAG4O7l-t1ubbwVAlPnDKg',
      rating: 5,
      user: {
        id: 'W8UK02IDdRS2GL_66fuq6w',
        profile_url:
          'https://www.yelp.com/user_details?userid=W8UK02IDdRS2GL_66fuq6w',
        image_url:
          'https://s3-media3.fl.yelpcdn.com/photo/iwoAD12zkONZxJ94ChAaMg/o.jpg',
        name: 'Ella A.'
      },
      text: 'Went back again to this place since the last time i visited the bay area 5 months ago, and nothing has changed. Still the sketchy Mission, Still the cashier...',
      time_created: '2016-08-29 00:41:13',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=hp8hAJ-AnlpqxCCu7kyCWA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: '1JNmYjJXr9ZbsfZUAgkeXQ',
      rating: 4,
      user: {
        id: 'rk-MwIUejOj6LWFkBwZ98Q',
        profile_url:
          'https://www.yelp.com/user_details?userid=rk-MwIUejOj6LWFkBwZ98Q',
        image_url: null,
        name: 'Yanni L.'
      },
      text: 'The "restaurant" is inside a small deli so there is no sit down area. Just grab and go.\n\nInside, they sell individually packaged ingredients so that you can...',
      time_created: '2016-09-28 08:55:29',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=fj87uymFDJbq0Cy5hXTHIA&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    },
    {
      id: 'SIoiwwVRH6R2s2ipFfs4Ww',
      rating: 4,
      user: {
        id: 'rpOyqD_893cqmDAtJLbdog',
        profile_url:
          'https://www.yelp.com/user_details?userid=rpOyqD_893cqmDAtJLbdog',
        image_url: null,
        name: 'Suavecito M.'
      },
      text: 'Dear Mission District,\n\nI miss you and your many delicious late night food establishments and vibrant atmosphere.  I miss the way you sound and smell on a...',
      time_created: '2016-08-10 07:56:44',
      url: 'https://www.yelp.com/biz/la-palma-mexicatessen-san-francisco?hrid=m_tnQox9jqWeIrU87sN-IQ&adjust_creative=0sidDfoTIHle5vvHEBvF0w&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_reviews&utm_source=0sidDfoTIHle5vvHEBvF0w'
    }
  ],
  total: 3,
  possible_languages: ['en']
};
