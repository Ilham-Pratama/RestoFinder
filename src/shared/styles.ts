import { StyleSheet } from 'react-native';

export const ON_PRESS_BACKGROUND_COLOR = 'rgba(160, 160, 160, 0.2)';

const sharedStyles = StyleSheet.create({
  elevation: {
    shadowColor: 'black',
    shadowOffset: {
      height: 5,
      width: 5
    },
    elevation: 3,
    shadowOpacity: 0.1
  },

  flex1: {
    flex: 1
  },

  flexDirectionRow: {
    flexDirection: 'row'
  },

  alignItemsCenter: {
    alignItems: 'center'
  },

  justifyContentCenter: {
    justifyContent: 'center'
  },
  justifyContentSpaceBetween: {
    justifyContent: 'space-between'
  },

  padding1: {
    padding: 5
  },
  padding2: {
    padding: 10
  },
  padding3: {
    padding: 15
  },
  padding4: {
    padding: 20
  },

  paddingHorizontal1: {
    paddingHorizontal: 5
  },
  paddingHorizontal2: {
    paddingHorizontal: 10
  },
  paddingHorizontal3: {
    paddingHorizontal: 15
  },
  paddingHorizontal4: {
    paddingHorizontal: 20
  },

  paddingVertical1: {
    paddingVertical: 5
  },
  paddingVertical2: {
    paddingVertical: 10
  },
  paddingVertical3: {
    paddingVertical: 15
  },
  paddingVertical4: {
    paddingVertical: 20
  },

  marginTop1: {
    marginTop: 5
  },
  marginTop2: {
    marginTop: 10
  },
  marginTop3: {
    marginTop: 15
  },
  marginTop4: {
    marginTop: 20
  },

  marginBottom1: {
    marginBottom: 5
  },
  marginBottom2: {
    marginBottom: 10
  },
  marginBottom3: {
    marginBottom: 15
  },
  marginBottom4: {
    marginBottom: 20
  },

  marginRight1: {
    marginRight: 5
  },
  marginRight2: {
    marginRight: 10
  },
  marginRight3: {
    marginRight: 15
  },
  marginRight4: {
    marginRight: 20
  },

  marginHorizontal1: {
    marginHorizontal: 5
  },
  marginHorizontal2: {
    marginHorizontal: 10
  },
  marginHorizontal3: {
    marginHorizontal: 15
  },
  marginHorizontal4: {
    marginHorizontal: 20
  },

  marginVertical1: {
    marginVertical: 5
  },
  marginVertical2: {
    marginVertical: 10
  },
  marginVertical3: {
    marginVertical: 15
  },
  marginVertical4: {
    marginVertical: 20
  },

  textBold: {
    fontWeight: 'bold'
  },

  textSemiBold: {
    fontWeight: '700'
  }
});

export default sharedStyles;
