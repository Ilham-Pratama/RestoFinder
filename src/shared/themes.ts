import { Theme } from '@react-navigation/native';
import { DarkTheme } from '@react-navigation/native';

export const AppLightTheme: Theme = {
  dark: false,
  colors: {
    primary: '#c3f229',
    background: '#F2F2F2',
    card: '#FFF',
    text: '#1C1C1E',
    border: '#E2FFA7',
    notification: '#C1FC4C'
  }
};

export const AppDarkTheme: Theme = {
  dark: true,
  colors: {
    ...DarkTheme.colors,
    primary: '#3D3D3D',
    card: '#1A1A1A',
    background: '#121212'
  }
};
