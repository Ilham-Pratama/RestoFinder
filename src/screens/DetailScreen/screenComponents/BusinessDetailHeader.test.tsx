import React from 'react';
import { render } from '@testing-library/react-native';
import BusinessDetailHeader from './BusinessDetailHeader';
import { BUSINESS_DUMMY } from 'shared/mocks';

describe('BusinessDetailHeader', () => {
  it('renders BusinessDetailHeader', () => {
    const component = render(<BusinessDetailHeader {...BUSINESS_DUMMY} />);

    expect(component.toJSON()).toMatchSnapshot();
    expect(component.getAllByRole('image')).toHaveLength(8);
    expect(component.getByRole('header').props).toHaveProperty(
      'children',
      BUSINESS_DUMMY.name
    );
    expect(
      component.getAllByText(`${BUSINESS_DUMMY.review_count}`)
    ).toBeTruthy();
    expect(component.getByRole('none').props.colors).toHaveLength(6);
  });
});
