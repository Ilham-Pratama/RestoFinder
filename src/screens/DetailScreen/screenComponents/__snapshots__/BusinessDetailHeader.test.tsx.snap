// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessDetailHeader renders BusinessDetailHeader 1`] = `
<View>
  <RCTScrollView
    accessibilityRole="list"
    bounces={false}
    data={
      Array [
        "https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg",
        "https://s3-media4.fl.yelpcdn.com/bphoto/FmXn6cYO1Mm03UNO5cbOqw/o.jpg",
        "https://s3-media4.fl.yelpcdn.com/bphoto/HZVDyYaghwPl2kVbvHuHjA/o.jpg",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "height": 250,
          "overflow": "hidden",
        },
      ]
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "uri": "https://s3-media2.fl.yelpcdn.com/bphoto/CPc91bGzKBe95aM5edjhhQ/o.jpg",
            }
          }
          style={
            Object {
              "width": 750,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "uri": "https://s3-media4.fl.yelpcdn.com/bphoto/FmXn6cYO1Mm03UNO5cbOqw/o.jpg",
            }
          }
          style={
            Object {
              "width": 750,
            }
          }
        />
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            Object {
              "uri": "https://s3-media4.fl.yelpcdn.com/bphoto/HZVDyYaghwPl2kVbvHuHjA/o.jpg",
            }
          }
          style={
            Object {
              "width": 750,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <BVLinearGradient
    accessibilityRole="none"
    colors={
      Array [
        0,
        0,
        0,
        1560281088,
        2701131776,
        3456106496,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "justifyContent": "flex-end",
          "left": 0,
          "position": "absolute",
          "right": 0,
        },
        Object {
          "paddingVertical": 10,
        },
        Object {
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    <Text
      accessibilityRole="header"
      style={
        Object {
          "color": "#f4f4f4",
          "fontSize": 30,
          "fontWeight": "800",
        }
      }
    >
      Gary Danko
    </Text>
    <RCTScrollView
      ListFooterComponent={
        <View
          style={
            Object {
              "marginHorizontal": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#f4f4f4",
              }
            }
          >
            5296
          </Text>
        </View>
      }
      data={
        Array [
          1,
          2,
          3,
          4,
          5,
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "marginVertical": 5,
          },
        ]
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#c3f229"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={576}
              width={20}
            >
              <RNSVGGroup
                fill={4291031593}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#c3f229"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={576}
              width={20}
            >
              <RNSVGGroup
                fill={4291031593}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#c3f229"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={576}
              width={20}
            >
              <RNSVGGroup
                fill={4291031593}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#c3f229"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={576}
              width={20}
            >
              <RNSVGGroup
                fill={4291031593}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <RNSVGSvgView
              accessibilityRole="image"
              align="xMidYMid"
              bbHeight={20}
              bbWidth={20}
              fill="#c3f229"
              focusable={false}
              height={20}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 20,
                    "width": 20,
                  },
                ]
              }
              vbHeight={512}
              vbWidth={576}
              width={20}
            >
              <RNSVGGroup
                fill={4291031593}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  d="M463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7zM288 376.4L288.1 376.3L399.7 435.9L378.4 309.6L469.2 219.8L343.8 201.4L288.1 86.85L288 87.14V376.4z"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "marginHorizontal": 5,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#f4f4f4",
                }
              }
            >
              5296
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
